pkgs/applications/science/biology/neuron/default.nix

```diff
@@ -1,94 +1,108 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchurl
-, pkg-config
-, automake
-, autoconf
-, libtool
-, ncurses
 , readline
-, which
-, python ? null
-, useMpi ? false
+, xorg
 , mpi
-, iv
+, cmake
+, bison
+, flex
+, git
+, perl
+, gsl
+, xcbuild
+, python3
+, useMpi ? false
+, useIv ? true
+, useCore ? false
+, useRx3d ? false
 }:
 
+
 stdenv.mkDerivation rec {
-  pname = "neuron${lib.optionalString useMpi "-mpi"}";
-  version = "7.5";
+  pname = "neuron";
+  version = "8.2.1";
 
-  nativeBuildInputs = [ which pkg-config automake autoconf libtool ];
-  buildInputs = [ ncurses readline python iv ]
-    ++ lib.optional useMpi mpi;
+  # format is for pythonModule conversion
+  format = "other";
 
-  src = fetchurl {
-    url = "https://www.neuron.yale.edu/ftp/neuron/versions/v${version}/nrn-${version}.tar.gz";
-    sha256 = "0f26v3qvzblcdjg7isq0m9j2q8q7x3vhmkfllv8lsr3gyj44lljf";
-  };
+  nativeBuildInputs = [
+    cmake
+    bison
+    flex
+    git
+  ] ++ lib.optional useCore [ perl gsl ]
+  ++ lib.optional stdenv.isDarwin [ xcbuild ];
 
-  patches = (lib.optionals (stdenv.isDarwin) [ ./neuron-carbon-disable.patch ]);
+  buildInputs = lib.optional useIv [
+    xorg.libX11
+    xorg.libX11.dev
+    xorg.libXcomposite
+    xorg.libXcomposite.dev
+    xorg.libXext
+    xorg.libXext.dev
```

Comment:
You are sure about requiring both variants? the one without .dev should be enough.

Reply:
I have minimized the reqs, Xcomposite and Xext require dev, X11 doesn't. Regular Xcomposite and Xext can be removed safely and have in the latest commit